lib/membrane_ffmpeg_swscale/scaler/scaler.ex

```diff
@@ -70,22 +70,26 @@ defmodule Membrane.FFmpeg.SWScale.Scaler do
       ) do
     case Native.scale(payload, use_shm?, native_state) do
       {:ok, frame} ->
-        buffer = [buffer: {:output, %{buffer | payload: frame}}]
-        {{:ok, buffer}, state}
+        {[buffer: {:output, %{buffer | payload: frame}}], state}
 
       {:error, reason} ->
-        {{:error, reason}, state}
+        raise "Scaling failed: #{reason}"
     end
   end
 
   @impl true
-  def handle_caps(:input, %RawVideo{width: width, height: height} = caps, _context, state) do
+  def handle_stream_format(
+        :input,
+        %RawVideo{width: width, height: height} = stream_format,
+        _context,
+        state
+      ) do
     case Native.create(width, height, state.output_width, state.output_height) do
       {:ok, native_state} ->
-        caps = %{caps | width: state.output_width, height: state.output_height}
+        stream_format = %{stream_format | width: state.output_width, height: state.output_height}
         state = %{state | native_state: native_state}
 
-        {{:ok, caps: {:output, caps}}, state}
+        {[stream_format: {:output, stream_format}], state}
 
       {:error, reason} ->
         raise(RuntimeError, reason)
```

Comment:
1. reason must be string
2. `RuntimeError` is the default
```suggestion
        raise inspect(reason)
```